webapp/src/main/java/life/qbic/usermanagement/registration/EmailConfirmationLinkSupplier.java

```diff
@@ -11,28 +13,32 @@
 @Service
 public class EmailConfirmationLinkSupplier {
 
+  private final String protocol;
+
   private final String host;
 
-  private final String port;
+  private final int port;
 
-  private final String loginEndpoint;
+  private final String emailConfirmationEndpoint;
 
   private final String emailConfirmationParameter;
 
-  public EmailConfirmationLinkSupplier(@Value("${host.name}") String host,
-      @Value("${server.port}") String port,
-      @Value("${login-endpoint}") String loginEndpoint,
+  public EmailConfirmationLinkSupplier(
+      @Value("${service.host.protocol}") String protocol,
+      @Value("${service.host.name}") String host,
+      @Value("${server.port}") int port,
+      @Value("${email-confirmation-endpoint}") String loginEndpoint,
       @Value("${email-confirmation-parameter}") String emailConfirmationParameter) {
+    this.protocol = protocol;
     this.host = host;
     this.port = port;
-    this.loginEndpoint = loginEndpoint;
+    this.emailConfirmationEndpoint = loginEndpoint;
     this.emailConfirmationParameter = emailConfirmationParameter;
   }
 
-  public String emailConfirmationUrl(String userId) {
-    String hostAddress = String.join(":", host, port);
-    String params = String.join("=", emailConfirmationParameter,  userId);
-    String endpointWithParams = String.join("?", loginEndpoint, params);
-    return String.join("/", hostAddress, endpointWithParams);
+  public String emailConfirmationUrl(String userId) throws MalformedURLException {
```

Comment:
Same as passwordResetURL:
Very good idea, however this should also be tested in the Spec(e.g. "htt" as protocol)

Reply:
Not needed, that is part of the java.net.URL and it will work ;)